### DISCUSSION TITLE
Include Path error in VS Code

### DISCUSSION THREAD

**Discussion by rishi-pathak** [microsoft/vscode-cpptools#9018] [Q&A]
Hi, I am new to coding and currently struggling with VS Code include path configurations.  Would really appreciate any possible help.

I tried running a code but I got an error saying 
 " #include errors detected. Please update your includePath. Squiggles are disabled for this translation unit (C:\Users\csris\OneDrive\Desktop\Code\Fibonacci)." 
and,
  cannot open source file "iostream"

I tried configuring my include path and compiler path but nothing is working. 

Here is the code I was trying to run
![● Fibonacci - Code - Visual Studio Code 13-03-2022 17_58_41](https://user-images.githubusercontent.com/101506752/158059364-d002577b-70c3-48b9-bbe8-95a484e91612.png)

here is my config

![Include Path error in VS Code · Discussion #9018 · microsoft_vscode-cpptools - Google Chrome 13-03-2022 17_59_49](https://user-images.githubusercontent.com/101506752/158059376-0d95163c-77d2-4844-bd2d-0d51cff5cfcc.png)

**Comment by sean-mcmanus** [comment]
Remove the system include from your includePath setting. Does that fix your issue?

**Reply by rishi-pathak** [reply]
I tried it , but it didn't work

**Reply by sean-mcmanus** [reply]
What version of the C/C++ extension are you using? It may be failing to query your compiler. Can you run the C/C++: Log Diagnostics command and see what compiler path and includes are being reported?

**Reply by rishi-pathak** [reply]
> What version of the C/C++ extension are you using? It may be failing to query your compiler. Can you run the C/C++: Log Diagnostics command and see what compiler path and includes are being reported?


v1.8.4

**Reply by sean-mcmanus** [reply]
If you can check your C/C++ or C/C++ Diagnostics logging that could help figure out why it appears to be failing to query your compiler for  system includes.

**Reply by sean-mcmanus** [reply]
Does the comment at https://github.com/microsoft/vscode-cpptools/issues/8987#issuecomment-1062166938 help?

**Reply by rishi-pathak** [reply]
> Does the comment at [#8987 (comment)](https://github.com/microsoft/vscode-cpptools/issues/8987#issuecomment-1062166938) help?

So should I share the diagnostics logging?

**Reply by sean-mcmanus** [reply]
That could help if you're not able to diagnose what is wrong with the logging yourself.

**Reply by sean-mcmanus** [reply]
You should not have "C:\MINGW\LIB\GCC\MINGW32\6.3.0" in your include path.

**Reply by rishi-pathak** [reply]
> That could help if you're not able to diagnose what is wrong with the logging yourself.


ok here you go

-------- Diagnostics - 18/3/2022, 7:50:56 pm
Version: 1.8.4
Current Configuration:
{
    "name": "Win32",
    "includePath": [
        "${workspaceFolder}/**",
        "C:\\MinGW\\lib\\gcc\\mingw32\\6.3.0"
    ],
    "defines": [
        "_DEBUG",
        "UNICODE",
        "_UNICODE"
    ],
    "compilerPath": "C:\\MinGW\\bin\\g++.exe",
    "cStandard": "gnu11",
    "cppStandard": "gnu++14",
    "intelliSenseMode": "clang-x64",
    "compilerPathIsExplicit": true,
    "cStandardIsExplicit": true,
    "cppStandardIsExplicit": true,
    "intelliSenseModeIsExplicit": true,
    "compilerArgs": [],
    "mergeConfigurations": false,
    "browse": {
        "path": [
            "${workspaceFolder}/**",
            "C:\\MinGW\\lib\\gcc\\mingw32\\6.3.0"
        ],
        "limitSymbolsToIncludedHeaders": true
    }
}
Translation Unit Mappings:
[ C:\Users\csris\OneDrive\Desktop\Code\.vscode\Fibonacci ]:
    C:\Users\csris\OneDrive\Desktop\Code\.vscode\Fibonacci
Translation Unit Configurations:
[ C:\Users\csris\OneDrive\Desktop\Code\.vscode\Fibonacci ]:
    Process ID: 11692
    Memory Usage: 14 MB
    Compiler Path: C:\MinGW\bin\g++.exe
    Includes:
        C:\MINGW\LIB\GCC\MINGW32\6.3.0
        C:\MINGW\LIB\GCC\MINGW32\6.3.0\INCLUDE
        C:\MINGW\INCLUDE
        C:\MINGW\LIB\GCC\MINGW32\6.3.0\INCLUDE-FIXED
        C:\MINGW\MINGW32\INCLUDE
    Defines:
        _DEBUG
        UNICODE
        _UNICODE
    Standard Version: c11
    IntelliSense Mode: windows-gcc-x86
    Other Flags:
        --gcc
        --gnu_version=60300
        --header_only_fallback
Total Memory Usage: 14 MB

------- Workspace parsing diagnostics -------
Number of files discovered (not excluded): 3216

**Reply by rishi-pathak** [reply]
> You should not have "C:\MINGW\LIB\GCC\MINGW32\6.3.0" in your include path.

I read some articles and they told me to do so ; So what really should I have as I don't know I am just starting out vscode because my 9th grade just ended and wanted to acquire some new skills.

**Reply by sean-mcmanus** [reply]
You should not have that path added -- it could cause incorrect includes to be picked up. What do you get when you remove that?

Whatever articles you're reading may be outdated or incorrect.

Also, it's highly recommended to use a newer compiler than gcc 6.3. The msys2 with gcc 11.2: see https://code.visualstudio.com/docs/cpp/config-mingw .

Your log diagnostics system includes would look something like
```
    Compiler Path: C:\msys64\mingw64\bin\g++.exe
    Includes:
        C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0
        C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0\X86_64-W64-MINGW32
        C:\MSYS64\MINGW64\INCLUDE\C++\11.2.0\BACKWARD
        C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\11.2.0\INCLUDE
        C:\MSYS64\MINGW64\INCLUDE
        C:\MSYS64\MINGW64\LIB\GCC\X86_64-W64-MINGW32\11.2.0\INCLUDE-FIXED
        C:\MSYS64\MINGW64\X86_64-W64-MINGW32\INCLUDE
```
